src/main/scala/gitbucket/core/model/activity/comment/CommentActivityInfo.scala

```diff
@@ -0,0 +1,95 @@
+package gitbucket.core.model.activity.comment
+
+import java.util.UUID
+
+import gitbucket.core.model.Activity
+import gitbucket.core.model.Profile.currentDate
+import gitbucket.core.model.activity.BaseActivityInfo
+
+sealed trait CommentActivityInfo extends BaseActivityInfo {
```

Comment:
Even if we have a base trait for specific types of activity information, each sub case classes need to have conductor arguments and we need to implement conversion to `Activity` in each class. Therefore, these base trait would be not much meaningful.

Reply:
Right, I understand what you're saying here. 

I think the benefit of this would be future ergonomics and extensibility. An alternative way of doing this would be to
* have each case class extend the `BaseActivityInfo` trait

This could remove the need for our base traits. That being said, I don't think this is the correct approach. I could see a point in the future where we want to have a function that consumes some activity. We could have a function like `logOnlyCommentInfo`. With our first approach of having the base trait, we'd have something like

```Scala
def logOnlyCommentInfo(info: BaseActivityInfo): Unit = info match {
  case 1 => ...
   ...
  case n =>
}
```
of course we could simplify this by using the wildcard operator `_` for pattern matching, but I think that could sometimes be error prone, with our approach of having a sealed trait up top, we could just have
```Scala
def logOnlyCommentInfo(info: CommentActivityInfo): Unit = info match {
  case 1 => ..
  case "last sub type of CommentActivityInfo" => ...
}
```
I would also argue that this is much safer. We get a compile-time error if we fail to exhaustively provide a case for all subtypes of `CommentActivityInfo`.

What are your thoughts?

Reply:
> I think the benefit of this would be future ergonomics and extensibility.

If we have a plan to have such narrower logging methods, intermediate traits like `CommentActivityInfo` would be beneficial in the near future. I understand the technical benefit of it but I'm wondering whether there is such a possibility in our future.

Anyway, introducing intermediate traits doesn't have a downside while it may be beneficial in the future, so I'm not against introducing intermediate traits.

Reply:
Decided to remove intermediate traits, see https://github.com/gitbucket/gitbucket/pull/2526#discussion_r478752552